impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -240,7 +240,7 @@ static String getProxyName(String contextId, Class<?> proxiedBeanType, Set<? ext
 
     private static ProxyNameHolder createCompoundProxyName(String contextId, Bean<?> bean, TypeInfo typeInfo, StringBuilder name) {
         String className;
-        String proxyPackage = null;
+        String proxyPackage = typeInfo.getPackageNameForClass(bean.getBeanClass());
```

Comment:
Hmm, I think this can result in a mismatch of proxy package name with the first class name which will further down the line lead to what was fixed in https://issues.redhat.com/browse/WELD-2666

This is the reason why we had the current `if` condition inside the loop.

Also note that `bean.getBeanClass()` for producer method will, by CDI definition, return the class of the bean that declares the producer method. See https://github.com/eclipse-ee4j/cdi/blob/master/api/src/main/java/jakarta/enterprise/inject/spi/Bean.java#L37-L43
This is something you don't want because the class of the declaring bean can be in a different module than where the proxy needs to be created.

Reply:
I just updated the PR, could you please take another look?

Reply:
Also, I'm not very familiar with the code base but I outlined the case in the issue: https://issues.redhat.com/browse/WELD-2672

Reply:
I think I understand the general underlying issue and can create a similar reproducer but it would be without actual modular structure because we don't have that in tests (or I just don't know how to do it alongside current test setup).
For that reason, having a standalone reproducer app with actual modular setup would be very valuable.

As for the PR, I think you will still have the same problem of mismatch between proxy package and the first class which should result in CI failure again...

Reply:
I have some idea that I can try to implement this afternoon and see if it helps here. I'll report back once I know more.
If you could cobble the reproducer app together, that would help greatly.